configuration/configuration.go

```diff
@@ -89,36 +90,114 @@ const (
 	v *viper.Viper
+
+	// Service Account Configuration
+	sa *viper.Viper
 }
 
 // NewConfigurationData creates a configuration reader object using a configurable configuration file path
-func NewConfigurationData(configFilePath string) (*ConfigurationData, error) {
+func NewConfigurationData(mainConfigFile string, serviceAccountConfigFile string) (*ConfigurationData, error) {
 	c := ConfigurationData{
-		v: viper.New(),
+		v:  viper.New(),
+		sa: viper.New(),
 	}
+
+	// Set up the main configuration
 	c.v.SetEnvPrefix("AUTH")
 	c.v.AutomaticEnv()
 	c.v.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
 	c.v.SetTypeByDefaultValue(true)
 	c.setConfigDefaults()
 
-	if configFilePath != "" {
+	// Set up the service account configuration (stored in a separate config file)
+	c.sa.SetTypeByDefaultValue(true)
+
+	if mainConfigFile != "" {
 		c.v.SetConfigType("yaml")
-		c.v.SetConfigFile(configFilePath)
+		c.v.SetConfigFile(mainConfigFile)
 		err := c.v.ReadInConfig() // Find and read the config file
 		if err != nil {           // Handle errors reading the config file
 			return nil, errors.Errorf("Fatal error config file: %s \n", err)
 		}
 	}
+
+	// If a service account configuration file has been specified, check that it exists
+	if serviceAccountConfigFile != "" {
+		if _, err := os.Stat(serviceAccountConfigFile); os.IsNotExist(err) {
+			// If it does not exist, return an error
+			return nil, errors.Errorf("Specified service account configuration file does not exist: %s \n", err)
+		}
+	}
+
+	// If the service account configuration file has not been specified
+	// then we default to /etc/fabric8/service-account-secrets.conf
+	if serviceAccountConfigFile == "" {
+		if _, err := os.Stat(varDefaultServiceAccountConfig); !os.IsNotExist(err) {
+			serviceAccountConfigFile = varDefaultServiceAccountConfig
+		}
+	}
+
+	ex, err := os.Executable()
+	if err != nil {
+		panic(err)
+	}
+	exPath := filepath.Dir(ex)
+
+	// If we can't find it in the /etc/fabric8 directory, check the running directory
+	if serviceAccountConfigFile == "" {
+		checkPath := exPath + "/" + varServiceAccountConfigName
+		if _, err := os.Stat(checkPath); !os.IsNotExist(err) {
+			serviceAccountConfigFile = checkPath
+		}
+	}
+
+	// Otherwise check the /conf subdirectory of the running directory
+	if serviceAccountConfigFile == "" {
+		checkPath := exPath + "/conf/" + varServiceAccountConfigName
+		if _, err := os.Stat(checkPath); !os.IsNotExist(err) {
+			serviceAccountConfigFile = checkPath
+		}
+	}
+
+	// Otherwise as a last ditch attempt to locate the configuration, check the /conf subdirectory
+	// of the running directory's parent directory (../conf)
+	if serviceAccountConfigFile == "" {
+		checkPath := exPath + "/../conf/" + varServiceAccountConfigName
+		if _, err := os.Stat(checkPath); !os.IsNotExist(err) {
+			serviceAccountConfigFile = checkPath
+		}
+	}
+
+	// If the service account configuration file has not been specified
+	if serviceAccountConfigFile == "" {
```

Comment:
If the file still not found then we should return an error, shouldn't we? Otherwise load it:
```
if serviceAccountConfigFile == "" {
    return nil, errors.New(".....")
}
c.sa.SetConfigType("json")
c.sa.SetConfigFile(serviceAccountConfigFile)
```